FloripaPublicTransportation/RouteDetailViewController.swift

```diff
@@ -0,0 +1,195 @@
+    private var _departures: [Departure]? // stored property
+    private var departures: [Departure]? { // computed property
+        get { return _departures }
+        set {
+            guard newValue != nil else {
+                return
+            }
+            
+            _departures = newValue
+            
+            self.weekdayDepartures = _departures!.filter { d in
+                return d.calendar == .Weekday
+            }
+            
+            self.saturdayDepartures = _departures!.filter { d in
+                return d.calendar == .Saturday
+            }
+            
+            self.sundayDepartures = _departures!.filter { d in
+                return d.calendar == .Sunday
+            }
+        }
+    }
+    
+    // MARK: tableView.reloadData() logic
+    
+    private var _finishedLoadingStops = false
+    private var _finishedLoadingDepartures = false
+    var finishedLoadingStops: Bool {
+        get { return _finishedLoadingStops }
+        set {
+            _finishedLoadingStops = newValue
+            if finishedLoadingStops && finishedLoadingDepartures {
+                dispatch_async(dispatch_get_main_queue(), {
+                    self.tableView.reloadData()
+                    self.activityIndicator.stopAnimating()
+                })
+            }
+        }
+    }
+    var finishedLoadingDepartures: Bool {
+        get { return _finishedLoadingDepartures }
+        set {
+            _finishedLoadingDepartures = newValue
+            if finishedLoadingStops && finishedLoadingDepartures {
+                dispatch_async(dispatch_get_main_queue(), {
+                    self.tableView.reloadData()
+                    self.activityIndicator.stopAnimating()
+                })
+            }
+        }
+    }
+    
+    // MARK: Private constants definitions
+    
+    private let reuseIdentifier = "detailCell"
+    private let headerTitles: [String] = [
+        NSLocalizedString("List of streets within the route", comment: "The section header title to show in RouteDetailViewController above the streets that compose the route"),
+            NSLocalizedString("Weekday timetable", comment: "The section header title to show in RouteDetailViewController above the weekday timetable"),
+            NSLocalizedString("Saturday timetable", comment: "The section header title to show in RouteDetailViewController above the Saturday timetable"),
+            NSLocalizedString("Sunday timetable", comment: "The section header title to show in RouteDetailViewController above the Sunday timetable")]
+
+    // MARK: - View delegate
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        guard Connectivity.isConnectedToNetwork() else {
+            Connectivity.popNoConnectionAlert(self)
+            return
+        }
+        
+        self.activityIndicator.hidesWhenStopped = true
+        self.view.addSubview(self.activityIndicator)
```

Comment:
You mean layout constraints? Its position is defined in the routes property setter, which is accessed before viewDidLoad. I would have preferred to use constraints and I have programmed them, but then Xcode would claim about incompatible constraints, although there was nothing that could seem wrong with them. I have also tried them one by one - Xcode would reject them in anyway. So I have sticked to the frame calculation solution, which works and produces no warnings :)